lib/ts/recipe/emailpassword/components/library/formBase.tsx

```diff
@@ -151,8 +156,17 @@ export const FormBase: React.FC<FormBaseProps<any>> = (props) => {
                 }
             } catch (e) {
                 if (STGeneralError.isThisError(e)) {
-                    const message = e.message;
-                    props.onError(message);
+                    /**
+                     * Because status: GENERAL_ERROR has been replaced with
+                     * throwing STGeneralError, we need to handle this here too
+                     */
+                    if (unmounting.current.signal.aborted) {
+                        return;
+                    }
+
+                    handleFieldStateChange(fieldUpdates);
+
+                    props.onError(e.message);
                     return;
```

Comment:
In general, doing a return from checking `STGeneralError.isThisError(e)` is not good. What if `STGeneralError.isThisError(e)` is not true, and we just want to do  `props.onError("SOMETHING_WENT_WRONG_ERROR");` and continue to do something else?

So I would change this return everywhere to be like:
```
if (STGeneralError.isThisError(e)) {
   // handle general error
} else {
  // whatever the old code was doing in case of catching an exception
}
```